cli/template.go

```diff
@@ -363,16 +370,31 @@ func isMatchingPolicy(action schema.Action, policy *schema.Policy) bool {
 	return action.ID == policy.Action || policy.Action == "*" || (policy.Action == "read" && action.Method == "GET") || (policy.Action == "update" && action.Method == "POST")
 }
 
-func filterPolicies(principal string, policies []*schema.Policy) []*schema.Policy {
+func filterPolicies(principal string, scope []schema.TokenType, policies []*schema.Policy) []*schema.Policy {
 	var matchedPolicies []*schema.Policy
 	for _, policy := range policies {
-		if policy.Principal == principal {
-			matchedPolicies = append(matchedPolicies, policy)
+		if policy.Principal != principal {
+			continue
 		}
+
+		matchesScope := false
+		for _, tokenType := range scope {
```

Comment:
naming: we're mixing "scope" and "tokenType". I suggest `for _, scope := range scopes`